File: src/main/java/service/task/management/model/Employee.java
package service.task.management.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;
import javax.persistence.Entity;
import javax.persistence.Column;
import javax.persistence.Enumerated;
import javax.persistence.EnumType;
import javax.persistence.Table;

@Entity
@Getter
@Setter
@Table(name="employee")
public class Employee extends BaseModel {

    enum Role {
        ADMIN,
        MANAGER,
        USER
    }

    enum Gender {
        MALE,
        FEMALE,
        TRANSGENDER
    }

    @Column(name="employee_name")
    @JsonProperty("employee_name")
    private String employeeName;

    @Column(name = "employee_email")
    @JsonProperty("employee_email")
    private String employeeEmail;

    @Column(name = "password")
    @JsonProperty("password")
    private String password;

    @Column(name = "employee_role")
    @Enumerated(EnumType.STRING)
    @JsonProperty("employee_role")
    private Role employeeRole;

    @Column(name = "phone_no")
    @JsonProperty("phone_no")
    private Long phoneNo;

    @Column(name = "employee_address")
    @JsonProperty("employee_address")
    private String employeeAddress;

    @Column(name = "pin_code")
    @JsonProperty("pin_code")
    private int pinCode;

    @Column(name = "gender")
    @JsonProperty("gender")
    private Gender gender;
}

File: src/main/kotlin/service/task/management/model/Employee.kt
package service.task.management.model

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.NoArgsConstructor
import javax.persistence.Entity
import javax.persistence.Column
import javax.persistence.Enumerated
import javax.persistence.EnumType
import javax.persistence.Table

@NoArgsConstructor
@Entity
@Table(name = "employee")
data class Employee (

    @Column(name = "employee_name")
    @JsonProperty("employee_name")
    val employeeName: String? = null,

    @Column(name = "employee_email", unique = true)
    @JsonProperty("employee_email")
    val employeeEmail: String? = null,

    @Column(name = "password")
    @JsonProperty("password")
    var password: String? = null,

    @Column(name = "employee_role")
    @Enumerated(EnumType.STRING)
    @JsonProperty("employee_role")
    val employeeRole: Role? = null,

    @Column(name = "phone_no", unique = true)
    @JsonProperty("phone_no")
    var phoneNo: Long? = null,

    @Column(name = "employee_address")
    @JsonProperty("employee_address")
    var employeeAddress: String? = null,

    @Column(name = "pin_code")
    @JsonProperty("pin_code")
    var pinCode: Int? = null,

    @Column(name = "gender")
    @JsonProperty("gender")
    val gender: Gender? = null
) : BaseModel()

enum class Role {
    ADMIN, MANAGER, USER
}

enum class Gender {
    MALE, FEMALE, TRANSGENDER
}